tuf/tuf.go

```diff
@@ -202,6 +223,47 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey, before
 
 	tr.keysDB.AddRole(role)
 
+	utils.RemoveUnusedKeys(p)
+
+	return nil
+}
+
+// DeleteDelegation removes a delegated targets role from its parent
+// targets object. It also deletes the delegation from the snapshot.
+// DeleteDelegation will only make use of the role Name field.
+func (tr *Repo) DeleteDelegation(role data.Role) error {
```

Comment:
Removing a delegation is essentially orphaning it by removing it from the
parent, such that it won't be visible when traversing the repository via
the top level targets file. That naturally cuts off all the children of the
delegation too.

On Mon, Dec 14, 2015 at 11:51 PM, Ying Li <notifications@github.com> wrote:

> In tuf/tuf.go
> <https://github.com/docker/notary/pull/354#discussion_r47605867>:
>
> > @@ -202,6 +223,47 @@ func (tr *Repo) UpdateDelegations(role *data.Role, keys []data.PublicKey, before
> >
> >  	tr.keysDB.AddRole(role)
> >
> > +	utils.RemoveUnusedKeys(p)
> > +
> > +	return nil
> > +}
> > +
> > +// DeleteDelegation removes a delegated targets role from its parent
> > +// targets object. It also deletes the delegation from the snapshot.
> > +// DeleteDelegation will only make use of the role Name field.
> > +func (tr *Repo) DeleteDelegation(role data.Role) error {
>
> Should this also remove any child roles? Or error maybe if a child role is
> then orphaned?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/notary/pull/354/files#r47605867>.
>